internal/controllers/compute/machine_ephemeralvolume_controller.go

```diff
@@ -71,19 +72,29 @@ func (r *MachineEphemeralVolumeReconciler) ephemeralMachineVolumeByName(machine
 				Labels:      ephemeral.VolumeTemplate.Labels,
 				Annotations: maps.Clone(ephemeral.VolumeTemplate.Annotations),
 			},
-			Spec: ephemeral.VolumeTemplate.Spec,
+			Spec: ephemeral.VolumeTemplate.Spec.VolumeSpec,
 		}
 		annotations.SetDefaultEphemeralManagedBy(volume)
-		_ = ctrl.SetControllerReference(machine, volume, r.Scheme())
+		if machineVolume.Ephemeral.VolumeTemplate.Spec.ReclaimPolicy != storagev1alpha1.Retain {
```

Comment:
In case we use `Retain` policy for ephemeral `Volumes` and we don't set the controller reference, there will be no finalizer preventing the deletion of a Volume. We should prohibit the deletion of an ephemeral volume at all times as  long as the `Machine` object exists. One idea would be to always set the controller reference and remove it in the case of a `Machine` deletion.

Reply:
We might can utilize `RESTDeleteStrategy` in  `/k8s.io/apiserver@v0.30.7/pkg/registry/rest/delete.go`

Reply:
As discussed offline, If we want to handle removal of OwnerRef in registry using RESTDeleteStrategy, we are not adding ReclaimPolicy directly under VolumeSpec we have created wrapper around VolumeTemplateSpec as we didn't want to set this field for volumes other than ephemeral. so will not have reference to ReclaimPolicy in volme strategy. 
So handling removal of controller reference in machine controller before deleting machine.